src/load_inputs/load_generators_data.jl

```diff
@@ -213,5 +213,280 @@ function load_generators_data!(setup::Dict, path::AbstractString, inputs_gen::Di
+"""
+function load_vre_stor_data!(setup::Dict, path::AbstractString, inputs_gen::Dict, gen_in::DataFrame)
+	
+	error_strings = String[]
+	dfGen = inputs_gen["dfGen"]
+	inputs_gen["VRE_STOR"] = "VRE_STOR" in names(gen_in) ? gen_in[gen_in.VRE_STOR.==1,:R_ID] : Int[]
+
+	# Check if VRE-STOR resources exist
+	if !isempty(inputs_gen["VRE_STOR"])
+
+		# Check input data format
+		vre_stor_errors = check_vre_stor_validity(gen_in, setup)
+		append!(error_strings, vre_stor_errors)
+
+		vre_stor_in = DataFrame(CSV.File(joinpath(path,"Vre_and_stor_data.csv"), header=true), copycols=true)
+
+		## Defining all sets
+
+		# Solar PV Resources
+		inputs_gen["VS_SOLAR"] = vre_stor_in[(vre_stor_in.SOLAR.!=0),:R_ID]
+
+		# DC Resources
+		inputs_gen["VS_DC"] = union(vre_stor_in[vre_stor_in.STOR_DC_DISCHARGE.>=1,:R_ID], vre_stor_in[vre_stor_in.STOR_DC_CHARGE.>=1,:R_ID], vre_stor_in[vre_stor_in.SOLAR.!=0,:R_ID])
+
+		# Wind Resources
+		inputs_gen["VS_WIND"] = vre_stor_in[(vre_stor_in.WIND.!=0),:R_ID]
+
+		# Storage Resources
+		split_storage_resources!(vre_stor_in, inputs_gen, setup)
+
+		# Set of all VRE-STOR resources eligible for new solar capacity
+		inputs_gen["NEW_CAP_SOLAR"] = intersect(dfGen[dfGen.New_Build.==1,:R_ID], vre_stor_in[vre_stor_in.SOLAR.!=0,:R_ID], vre_stor_in[vre_stor_in.Max_Cap_Solar_MW.!=0,:R_ID])
+		# Set of all VRE_STOR resources eligible for solar capacity retirements
+		inputs_gen["RET_CAP_SOLAR"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID],  vre_stor_in[vre_stor_in.SOLAR.!=0,:R_ID], vre_stor_in[vre_stor_in.Existing_Cap_Solar_MW.>=0,:R_ID])
+		# Set of all VRE-STOR resources eligible for new wind capacity
+		inputs_gen["NEW_CAP_WIND"] = intersect(dfGen[dfGen.New_Build.==1,:R_ID], vre_stor_in[vre_stor_in.WIND.!=0,:R_ID], vre_stor_in[vre_stor_in.Max_Cap_Wind_MW.!=0,:R_ID])
+		# Set of all VRE_STOR resources eligible for wind capacity retirements
+		inputs_gen["RET_CAP_WIND"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], vre_stor_in[vre_stor_in.WIND.!=0,:R_ID], vre_stor_in[vre_stor_in.Existing_Cap_Wind_MW.>=0,:R_ID])
+		# Set of all VRE-STOR resources eligible for new inverter capacity
+		inputs_gen["NEW_CAP_DC"] = intersect(dfGen[dfGen.New_Build.==1,:R_ID], vre_stor_in[vre_stor_in.Max_Cap_Inverter_MW.!=0,:R_ID], inputs_gen["VS_DC"])
+		# Set of all VRE_STOR resources eligible for inverter capacity retirements
+		inputs_gen["RET_CAP_DC"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], vre_stor_in[vre_stor_in.Existing_Cap_Inverter_MW.>=0,:R_ID], inputs_gen["VS_DC"])
+		# Set of all storage resources eligible for new energy capacity
+		inputs_gen["NEW_CAP_STOR"] = intersect(dfGen[dfGen.New_Build.==1,:R_ID], dfGen[dfGen.Max_Cap_MWh.!=0,:R_ID], inputs_gen["VS_STOR"])
+		# Set of all storage resources eligible for energy capacity retirements
+		inputs_gen["RET_CAP_STOR"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], dfGen[dfGen.Existing_Cap_MWh.>=0,:R_ID], inputs_gen["VS_STOR"])
```

Comment:
The sets `dfGen[dfGen.New_Build.==1,:R_ID]` and `dfGen[dfGen.New_Build.==-1,:R_ID]` are used many times here. Please name these something else to reduce duplication of effort (and anticipate a future PR where the interface to NEW_BUILD changes).